cmd/zoekt-archive-index/ignore_test.go

```diff
@@ -0,0 +1,138 @@
+package main
+
+import (
+	"context"
+	"io/ioutil"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/google/zoekt"
+	"github.com/google/zoekt/build"
+	"github.com/google/zoekt/query"
+	"github.com/google/zoekt/shards"
+)
+
+type ignoreTest struct {
+	ignorePattern string
+	wantFiles     int
+}
+
+func TestIgnore(t *testing.T) {
+	tests := []ignoreTest{
+		{
+			ignorePattern: "D0",
+			wantFiles:     5,
+		},
+		{
+			ignorePattern: "D3",
+			wantFiles:     3,
+		},
+		{
+			ignorePattern: "/D3",
+			wantFiles:     3,
+		},
+		{
+			ignorePattern: "/D3/",
+			wantFiles:     3,
+		},
+		{
+			ignorePattern: "D3/",
+			wantFiles:     3,
+		},
+		{
+			ignorePattern: "D30",
+			wantFiles:     6,
+		},
+		{
+			ignorePattern: "D1\n # D3",
+			wantFiles:     5,
+		},
+		{
+			ignorePattern: "D1\nD3",
+			wantFiles:     2,
+		},
+		{
+			ignorePattern: "D1\n\nD3",
+			wantFiles:     2,
+		},
+		{
+			ignorePattern: "D0\nD1\nD2\nD3",
+			wantFiles:     0,
+		},
+	}
+	for _, format := range []string{"tar", "tgz", "zip"} {
+		for _, tt := range tests {
+			testIgnore(t, format, tt)
+		}
+	}
+}
+
+func testIgnore(t *testing.T, format string, test ignoreTest) {
+	fileSize := 10
+
+	indexdir, err := ioutil.TempDir("", "ignore_index")
+	if err != nil {
+		t.Fatalf("TempDir: %v", err)
+	}
+	//defer os.RemoveAll(indexdir)
```

Comment:
tests on CI fail if this line is not commented out (see previous commits). Locally tests run fine. Just looking at the GitHub action I cannot see what should be the difference.